modules/appeals_api/app/controllers/concerns/appeals_api/pdf_downloads.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'common/file_helpers'
+require 'pdf_fill/filler'
+
+# rubocop:disable Metrics/ModuleLength
+module AppealsApi
+  module PdfDownloads
+    extend ActiveSupport::Concern
+
+    included do
+      def render_appeal_pdf_download(appeal, filename)
+        if expired?(appeal)
+          # If the veteran_icn is available, we can check it against the current request's ICN header to return a 404
+          # if it does not match, but otherwise, without auth_headers, we can't tell whether the current request is
+          # authorized, so in that case we return a specific error message regardless of authorization status.
+          if appeal.veteran_icn && appeal.veteran_icn != request.headers['X-VA-ICN']
+            raise ActiveRecord::RecordNotFound
+          else
+            return render_pdf_download_expired(appeal)
+          end
+        end
+
+        # We choose to return a 404 when PII in headers doesn't match the Appeal's PII so that we don't reveal the
+        # existence of records that the user can't access
+        raise ActiveRecord::RecordNotFound unless download_authorized?(appeal)
+
+        return render_pdf_download_not_ready(appeal) unless submitted?(appeal)
+
+        pdf_path = PdfDownloads.watermark(
+          AppealsApi::PdfConstruction::Generator.new(appeal, pdf_version: appeal.pdf_version).generate,
+          filename
+        )
+
+        send_file(pdf_path, type: 'application/pdf; charset=utf-8', filename:)
+      end
+    end
+
+    # Creates a copy of the input PDF with a watermark on each page
+    def self.watermark(input_path, output_path = "#{Common::FileHelpers.random_file_path}.pdf")
+      num_pages = PDF::Reader.new(input_path).pages.length
+      stamp_path = "#{Common::FileHelpers.random_file_path}-watermark.pdf"
+
+      Prawn::Document.generate(stamp_path, margin: [10, 10]) do |pdf|
+        num_pages.times do
+          pdf.transparent(0.25) do
+            pdf.text_box WATERMARK_MARKUP,
+                         align: :center,
+                         inline_format: true,
+                         rotate: 45,
+                         rotate_around: :center,
+                         size: 52,
+                         valign: :center,
+                         overflow: :shrink_to_fit
+          end
+          pdf.start_new_page # Final extra page won't be added to output
+        end
+      end
+
+      PdfFill::Filler::PDF_FORMS.multistamp(input_path, stamp_path, output_path)
+      FileUtils.rm_f(stamp_path)
+
+      output_path
+    end
+
+    UNSUBMITTED_STATUSES = %w[pending submitting error].freeze
+    WATERMARK_MARKUP = '<b>DIGITALLY SUBMITTED<br>DO NOT FILE</b>'
```

Comment:
I used markup here because I was having trouble making the text bold without it - markup is enabled by including `inline_format: true` in the `text_box` options